targets/TARGET_STM/TARGET_STM32H7/TARGET_STM32H747xI/TARGET_PORTENTA_H7/system_clock_override.c

```diff
@@ -0,0 +1,304 @@
+uint8_t SetSysClock_PLL_HSI(void);
+#endif /* ((CLOCK_SOURCE) & USE_PLL_HSI) */
+
+/**
+  * @brief  Configures the System clock source, PLL Multiplier and Divider factors,
+  *               AHB/APBx prescalers and Flash settings
+  * @note   This function should be called only once the RCC clock configuration
+  *         is reset to the default reset state (done in SystemInit() function).
+  * @param  None
+  * @retval None
+  */
+
+void SetSysClock(void)
+{
+
+  bool lowspeed = false;
+#if defined(LOWSPEED) && (LOWSPEED == 1)
+  lowspeed = true;
+#endif
+
+#if ((CLOCK_SOURCE) & USE_PLL_HSE_EXTC)
+    /* 1- Try to start with HSE and external clock (MCO from STLink PCB part) */
+    if (SetSysClock_PLL_HSE(1, lowspeed) == 0)
+#endif
+    {
+#if ((CLOCK_SOURCE) & USE_PLL_HSE_XTAL)
+        /* 2- If fail try to start with HSE and external xtal */
+        if (SetSysClock_PLL_HSE(0, lowspeed) == 0)
+#endif
+        {
+#if ((CLOCK_SOURCE) & USE_PLL_HSI)
+            /* 3- If fail start with HSI clock */
+            if (SetSysClock_PLL_HSI() == 0)
+#endif
+            {
+                error("SetSysClock failed\n");
+            }
+        }
+    }
+}
+
+static const uint32_t _keep;
+bool isBootloader() {
+  return ((uint32_t)&_keep < 0x8040000);
+}
+
+bool isBetaBoard() {
+  uint8_t* bootloader_data = (uint8_t*)(0x801F000);
+  if (bootloader_data[0] != 0xA0 || bootloader_data[1] < 14) {
+    return true;
+  } else {
+    return (bootloader_data[10] == 27);
+  }
+}
+
+#if ( ((CLOCK_SOURCE) & USE_PLL_HSE_XTAL) || ((CLOCK_SOURCE) & USE_PLL_HSE_EXTC) )
+/******************************************************************************/
+/*            PLL (clocked by HSE) used as System clock source                */
+/******************************************************************************/
+uint8_t SetSysClock_PLL_HSE(uint8_t bypass, bool lowspeed)
+{
+    RCC_ClkInitTypeDef RCC_ClkInitStruct = {0};
+    RCC_OscInitTypeDef RCC_OscInitStruct = {0};
+    RCC_PeriphCLKInitTypeDef PeriphClkInitStruct = {0};
+
+    // If we are reconfiguring the clock, select CSI as system clock source to allow modification of the PLL configuration 
+    if (__HAL_RCC_GET_PLL_OSCSOURCE() == RCC_PLLSOURCE_HSE) {
+      RCC_ClkInitStruct.ClockType = RCC_CLOCKTYPE_SYSCLK;
+      RCC_ClkInitStruct.SYSCLKSource = RCC_SYSCLKSOURCE_CSI;
+      if (HAL_RCC_ClockConfig(&RCC_ClkInitStruct, FLASH_LATENCY_1) != HAL_OK)
+      {
+        return 0;
+      }
+    }
+
+    /* Enable oscillator pin */
+    __HAL_RCC_GPIOH_CLK_ENABLE();
+    GPIO_InitTypeDef  gpio_osc_init_structure;
+    gpio_osc_init_structure.Pin = GPIO_PIN_1;
+    gpio_osc_init_structure.Mode = GPIO_MODE_OUTPUT_PP;
+    gpio_osc_init_structure.Pull = GPIO_PULLUP;
+    gpio_osc_init_structure.Speed = GPIO_SPEED_FREQ_LOW;
+    HAL_GPIO_Init(GPIOH, &gpio_osc_init_structure);
+    HAL_Delay(10);
+    HAL_GPIO_WritePin(GPIOH, GPIO_PIN_1, 1);
+
+    /* Supply configuration update enable */
+#if HSE_VALUE == 27000000
```

Comment:
Why this value ?
I can see value is set to 25000000 in targets.json

Reply:
The oscillator on beta version of the boards was 27Mhz; we used to recognize it at runtime using a flash-based information "table" but the combination of no OTP area, big erase pages on H7 and the specific use case made me think it was better to keep as a non-default configuration value. 
It can be safely removed for the sake of clarity :slightly_smiling_face: